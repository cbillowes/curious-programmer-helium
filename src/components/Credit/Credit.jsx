import React, { Component } from "react"
import External from "../Link/ExternalLink"
const _ = require("lodash")

class Credit extends Component {
  render() {
    const { to, title } = this.props
    return (
      <div className="credit" key={`credit-${_.kebabCase(title)}`}>
        <span>{this.props.children}</span>
        <External to={to}>{title}</External>
      </div>
    )
  }
}

  export default Credit
