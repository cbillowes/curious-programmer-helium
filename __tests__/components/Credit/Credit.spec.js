import React from "react"
import renderer from "react-test-renderer"
import { mount } from "enzyme"
import Credit from "../../../src/components/Credit/Credit"

describe("a credit component", () => {
  it("should render", () => {
    const component = <Credit />
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  // it("should render contents", () => {
  //   const component = mount(<Tag to="/privacy-policy">Privacy Policy</Tag>)
  //   expect(component.html()).toEqual(`<a class="tag" href="/privacy-policy">Privacy Policy</a>`)
  // })
})