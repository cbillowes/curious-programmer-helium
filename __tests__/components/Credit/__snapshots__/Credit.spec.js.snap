// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a credit component should render 1`] = `
<div
  className="credit"
>
  <span />
  <a
    className=""
    rel="nofollow noopener noreferrer "
    target="_blank"
  />
</div>
`;
